esphome/components/atm90e32/atm90e32.h

```diff
@@ -51,53 +82,91 @@ class ATM90E32Component : public PollingComponent,
   void set_current_phases(int phases) { current_phases_ = phases; }
   void set_pga_gain(uint16_t gain) { pga_gain_ = gain; }
   void run_offset_calibrations();
+  void run_power_offset_calibrations();
+  void run_gain_calibrations();
   void clear_offset_calibrations();
+  void clear_power_offset_calibrations();
+  void clear_gain_calibrations();
   void set_enable_offset_calibration(bool flag) { enable_offset_calibration_ = flag; }
-  uint16_t calibrate_voltage_offset_phase(uint8_t /*phase*/);
-  uint16_t calibrate_current_offset_phase(uint8_t /*phase*/);
+  void set_enable_gain_calibration(bool flag) { enable_gain_calibration_ = flag; }
+  int16_t calibrate_offset(uint8_t phase, bool voltage);
+  int16_t calibrate_power_offset(uint8_t phase, bool reactive);
+  void set_reference_voltage(uint8_t phase, number::Number *ref_voltage) { ref_voltages_[phase] = ref_voltage; }
+  void set_reference_current(uint8_t phase, number::Number *ref_current) { ref_currents_[phase] = ref_current; }
+  float get_reference_voltage(uint8_t phase) {
+    return (phase >= 0 && phase < 3 && ref_voltages_[phase]) ? ref_voltages_[phase]->state : 120.0;  // Default voltage
+  }
+  float get_reference_current(uint8_t phase) {
+    return (phase >= 0 && phase < 3 && ref_currents_[phase]) ? ref_currents_[phase]->state : 5.0f;  // Default current
+  }
+  bool using_saved_calibrations_ = false;  // Track if stored calibrations are being used
+  void check_phase_status();
+  void check_freq_status();
+  void check_over_current();
+  void set_phase_status_text_sensor(uint8_t phase, text_sensor::TextSensor *sensor) {
+    this->phase_status_text_sensor_[phase] = sensor;
+  }
+  void set_freq_status_text_sensor(text_sensor::TextSensor *sensor) { this->freq_status_text_sensor_ = sensor; }
+  uint16_t calculate_voltage_threshold(int line_freq, uint16_t ugain, float multiplier);
   int32_t last_periodic_millis = millis();
 
+ private:
+  number::Number *ref_voltages_[3]{nullptr, nullptr, nullptr};
+  number::Number *ref_currents_[3]{nullptr, nullptr, nullptr};
```

Comment:
Same as the text sensor stuff
```suggestion
#ifdef USE_NUMBER
  number::Number *ref_voltages_[3]{nullptr, nullptr, nullptr};
  number::Number *ref_currents_[3]{nullptr, nullptr, nullptr};
#endif
```